runtime/vm/jfr.cpp

```diff
@@ -1020,7 +1021,7 @@ jfrThreadContextSwitchRate(J9VMThread *currentThread)
 				jfrEvent->switchRate = 0;
 			} else {
 				int64_t timeDelta = currentTime - jfrState->prevContextSwitchTimestamp;
-				jfrEvent->switchRate = ((double)(switches - jfrState->prevContextSwitches) / timeDelta) * 1e9;
+				jfrEvent->switchRate = ((float)(switches - jfrState->prevContextSwitches) / timeDelta) * 1e9f;
```

Comment:
Suggest this should do the division last:
```c
				jfrEvent->switchRate = (switches - jfrState->prevContextSwitches) * 1e9f / timeDelta;
```
What guarantees that `timeDelta` is not zero?

Reply:
Perhaps instead of `-1`, `prevContextSwitchTimestamp` should be initialized using `j9time_nano_time()` (on line 726):
```c
	vm->jfrState.prevContextSwitchTimestamp = j9time_nano_time();
```
There here, we avoid division by zero like this:
```c
			if (currentTime == jfrState->prevContextSwitchTimestamp) {
				jfrEvent->switchRate = 0f;
			} else {
				int64_t timeDelta = currentTime - jfrState->prevContextSwitchTimestamp;
				jfrEvent->switchRate = (switches - jfrState->prevContextSwitches) * 1e9f / timeDelta;
			}
```

Reply:
Sounds like a good idea and `prevContextSwitches` also need to be initialized using `omrsysinfo_get_number_context_switches`.